daemon/device_unix.go

```diff
@@ -0,0 +1,78 @@
+// +build !windows
+
+package daemon
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/containerd/containerd/pkg/userns"
+	"github.com/opencontainers/runc/libcontainer/devices"
+)
+
+// Testing dependencies
+var (
+	ioutilReadDir         = ioutil.ReadDir
+	usernsRunningInUserNS = userns.RunningInUserNS
+)
+
+// HostDevices returns all devices that can be found under /dev directory.
+// Based on HostDevices from runc (libcontainer/devices/device_unix.go).
+func HostDevices() ([]*devices.Device, error) {
+	return GetDevices("/dev")
+}
+
+// GetDevices recursively traverses a directory specified by path
+// and returns all devices found there.
+// Based on GetDevices from runc (libcontainer/devices/device_unix.go).
```

Comment:
Better to use a permalink here https://github.com/opencontainers/runc/blob/v1.0.0/libcontainer/devices/device_unix.go

Reply:
👆 Same for `HostDevices` above (permalink)